In [1]:
import os
import pandas as pd 
from pathlib import Path
import sys

PROJECT_ROOT = Path.cwd().parent   # notebooks/ -> project root
sys.path.append(str(PROJECT_ROOT))

data_paths_dict = { 
    "identity": os.path.join(PROJECT_ROOT, "data/raw/identity.csv"), 
    "transaction": os.path.join(PROJECT_ROOT, "data/raw/transaction.csv")
}

In [2]:
from src.preprocessing.clean import CleanData

In [10]:
features_dict, auc_dict = {}, {}
missing_threshold_list = [0.9, 0.95]
fi_threshold_list = [0.8, 0.85, 0.9, 0.95]

for missing_threshold in missing_threshold_list: 
    features_dict[missing_threshold], auc_dict[missing_threshold] = {}, {}
    for fi_threshold in fi_threshold_list: 
        pipeline = CleanData(
            data_paths_dict,
            missing_threshold,
            fi_threshold
        )
        df = pipeline.merge_data()
        df = pipeline.drop_dead_columns(df)
        _, df = pipeline.missingness_pvalues(df)
        df = pipeline.get_feature_importance(df)
        auc = pipeline.get_test_auc(df)
        features_dict[missing_threshold][fi_threshold] = df.shape[1]
        auc_dict[missing_threshold][fi_threshold] = auc

There are 34 features of zero importance
Keeping 60 features at 80.0% cumulative importance
There are 34 features of zero importance
Keeping 79 features at 85.0% cumulative importance
There are 34 features of zero importance
Keeping 103 features at 90.0% cumulative importance
There are 34 features of zero importance
Keeping 137 features at 95.0% cumulative importance
There are 29 features of zero importance
Keeping 58 features at 80.0% cumulative importance
There are 29 features of zero importance
Keeping 77 features at 85.0% cumulative importance
There are 29 features of zero importance
Keeping 101 features at 90.0% cumulative importance
There are 29 features of zero importance
Keeping 137 features at 95.0% cumulative importance


In [28]:
features_df = (
    pd.DataFrame.from_dict(features_dict, orient="index")
      .stack()
      .reset_index()
)

features_df.columns = ["missing_threshold", "fi_threshold", "features"]

auc_df = (
    pd.DataFrame.from_dict(auc_dict, orient="index")
      .stack()
      .reset_index()
)

auc_df.columns = ["missing_threshold", "fi_threshold", "auc"]

auc_feature_df = features_df.merge(
    right=auc_df, 
    how = "inner", 
    on = ["missing_threshold", "fi_threshold"]
)

auc_feature_df

,missing_threshold,fi_threshold,features,auc
0,0.90,0.80,61,0.882736
1,0.90,0.85,80,0.889971
2,0.90,0.90,104,0.890983
3,0.90,0.95,138,0.894055
4,0.95,0.80,59,0.869308
5,0.95,0.85,78,0.883045
6,0.95,0.90,102,0.891184
7,0.95,0.95,138,0.894360


The configuration with a missing-value threshold of 0.95 and a cumulative feature-importance threshold of 0.90 was selected because it achieved an AUC of 0.8912 using 102 features. Increasing the feature-importance threshold to 0.95 improved AUC by only 0.0032 while requiring 36 additional features, indicating that the 0.95/0.90 configuration provided the best trade-off between predictive performance and model complexity.

In [24]:
statement = "Hi Henry, noticed the Investment Risk quant role at Cbus. UNSW Stats Honours (First Class) grad here, built a systematic ASX equities strategy — dashboard.systematic-equities-alpha.com. What's the team's biggest focus, risk modelling or research? Gordon"

In [25]:
print(len(statement))

253


In [3]:
print(os.listdir(PROJECT_ROOT))

['.git', '.gitignore', '.vscode', 'data', 'notebooks', 'src']


In [5]:
import os, sys
print("On sys.path?", PROJECT_ROOT in sys.path)
print("src/__init__.py exists?", os.path.exists(os.path.join(PROJECT_ROOT, "src", "__init__.py")))
print("src/preprocessing/__init__.py exists?", os.path.exists(os.path.join(PROJECT_ROOT, "src", "preprocessing", "__init__.py")))
print(os.listdir(os.path.join(PROJECT_ROOT, "src")))

On sys.path? True
src/__init__.py exists? True
src/preprocessing/__init__.py exists? True
['data', 'pipeline.py', 'preprocessing', '__init__.py']


In [5]:
from src.preprocessing.clean import CleanData

ModuleNotFoundError: No module named 'src'